test/c/test_basis_translator.c

```diff
@@ -0,0 +1,165 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+int test_circuit_in_basis(void) {
+    // Create circuit
+    int result = Ok;
+    QkCircuit *circuit = qk_circuit_new(2, 0);
+    qk_circuit_gate(circuit, QkGate_H, (uint32_t[1]){0}, NULL);
+    qk_circuit_gate(circuit, QkGate_CX, (uint32_t[2]){0, 1}, NULL);
+
+    // Create Target compatible with only U gates, with global props.
+    QkTarget *target = qk_target_new(1);
+    qk_target_add_instruction(target, qk_target_entry_new(QkGate_H));
+    qk_target_add_instruction(target, qk_target_entry_new(QkGate_CX));
+
+    // Run pass
+    qk_transpiler_pass_standalone_basis_translator(circuit, target, 0);
+
+    size_t circuit_len = qk_circuit_num_instructions(circuit);
+    if (circuit_len != 2) {
+        result = EqualityError;
+        printf(
+            "The number of gates resulting from the translation is incorrect. Expected 2, got %lu",
+            circuit_len);
+        goto cleanup;
+    }
+
+    QkCircuitInstruction inst;
+    char *gate_names[2] = {"h", "cx"};
+    for (int idx = 0; idx < (int)circuit_len; idx++) {
```

Comment:
Why not just make `idx` as `size_t`?
```suggestion
    for (size_t idx = 0; idx < circuit_len; idx++) {
```